### PR TITLE
[Refactor](Nereids) Remove expression type.

### PR DESCRIPTION
**Pull request by wangshuo128**
# Proposed changes

Issue Number: close #xxx

`ExpressionType` is duplicated with Java class type info, so removed it.

## Problem Summary:

Describe the overview of changes.

## Checklist(Required)

1. Does it affect the original behavior: (Yes/No/I Don't know)
2. Has unit tests been added: (Yes/No/No Need)
3. Has document been added or modified: (Yes/No/No Need)
4. Does it need to update dependencies: (Yes/No)
5. Are there any changes that cannot be rolled back: (Yes/No)

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: fe/fe-core/src/main/java/org/apache/doris/nereids/trees/expressions/Regexp.java [hunk 1]
```diff
@@ -30,19 +30,14 @@
...
-        super(ExpressionType.REGEXP, left, right);
+        super(left, right, "regexp");
     }
 
     @Override
     public boolean nullable() throws UnboundException {
         return left().nullable();
     }
 
-    @Override
```
Comment (jackwener) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/trees/expressions/Regexp.java:41]: Why delete many `toString`

Comment (wangshuo128) [reply fe/fe-core/src/main/java/org/apache/doris/nereids/trees/expressions/Regexp.java:41]: It's pulled up to base class.

File: fe/fe-core/src/main/java/org/apache/doris/nereids/trees/expressions/StringRegexPredicate.java [hunk 2]
```diff
@@ -29,15 +29,21 @@
...
     /**
      * Constructor of StringRegexPredicate.
      *
-     * @param nodeType node type of expression
      * @param left     left child of string regex
      * @param right    right child of string regex
      */
-    public StringRegexPredicate(ExpressionType nodeType, Expression left, Expression right) {
-        super(nodeType, left, right);
+    public StringRegexPredicate(Expression left, Expression right, String symbol) {
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/trees/expressions/StringRegexPredicate.java:44]: nit: mod java doc parameter list

**Review by morrySnow — DISMISSED** [review]

File: fe/fe-core/src/main/java/org/apache/doris/nereids/trees/expressions/Arithmetic.java [hunk 3]
```diff
@@ -189,6 +165,24 @@ public int hashCode() {
 
     @Override
     public String toString() {
-        return toSql();
+        return stringBuilder(Object::toString);
```
Comment (924060929) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/trees/expressions/Arithmetic.java:168]: good refactor

**Review by englefly — APPROVED** [review]

File: fe/fe-core/src/main/java/org/apache/doris/nereids/trees/expressions/Arithmetic.java (hunk 3 above)
Comment (englefly) [reply fe/fe-core/src/main/java/org/apache/doris/nereids/trees/expressions/Arithmetic.java:168]: 为什么不抛未实现的异常，而是返回 java.lang.Object@d716361 这个样式的字符串？

**Review by jackwener — APPROVED** [review]

**Review by 924060929 — APPROVED** [review]
LGTM

**Pull request merged**